Fix fileno error on Windows (Serial bus)

MattWoodhead opened the pull request:
I was having issues with the Async example from the docs using the Serial interface: `io.UnsupportedOperation: fileno`

I looked back through the issues and found that several people have hit the fileno error over the years (#760, #877, #902). I have implemented the most recent fix from #1313 on the serial_bus.py interface and added a test case to the test_serial.py file.

zariiii9003 commented:
Thank you! Could you undo the Readme changes? The conda-forge version seems unmaintained.

MattWoodhead commented:
Sorry @zariiii9003, I have reverted the readme change. That must have snuck in when I rebased my local repository. I made a pull request to the conda-forge feedstock repository some time ago, but as you noted it appears to be unmainained.
